homeassistant/components/modbus/modbus.py

```diff
@@ -129,6 +130,16 @@ async def async_modbus_setup(
 ) -> bool:
     """Set up Modbus component."""
 
+    if DOMAIN in hass.data:  # pragma: no cover
```

Comment:
What is going to call `async_modbus_setup` for a second time? If you want `async_setup` to be called a second time, you need to define `async_reset_platform` https://github.com/home-assistant/core/blob/dev/homeassistant/helpers/reload.py#L77-L82

Reply:
Corrected, and also did a lot of searching/debugging in core. I hope it is to your liking now.